src/reason-parser/reason_parser.mly

```diff
@@ -3330,54 +3352,23 @@ expr_list:
 
 (* [x, y, z, ...n] --> ([x,y,z], Some n) *)
 expr_comma_seq_extension:
-  lseparated_nonempty_list(COMMA, expr_seq_item) COMMA?
+  lseparated_nonempty_list(COMMA, opt_spread(expr_optional_constraint)) COMMA?
   { match List.rev $1 with
     (* Check if the last expr has been spread with `...` *)
     | ((dotdotdot, e) as hd)::es ->
       let (es, ext) = match dotdotdot with
       | Some dotdotdotLoc -> (es, Some e)
       | None -> (hd::es, None)
       in
-      let exprList = List.map (fun (dotdotdot, e) -> match dotdotdot with
-      | Some dotdotdotLoc ->
-        (* If the `...` appears in any other location then the last,
-         * show a friendly, clear message explaining the consequences. *)
-        let msg = "Lists can only have one `...` spread, and at the end.
+      let msg = "Lists can only have one `...` spread, and at the end.
 Explanation: lists are singly-linked list, where a node contains a value and points to the next node. `[a, ...bc]` efficiently creates a new item and links `bc` as its next nodes. `[...bc, a]` would be expensive, as it'd need to traverse `bc` and prepend each item to `a` one by one. We therefore disallow such syntax sugar.
-Solution: directly use `concat`."
-        in
-        raise Reason_syntax_util.(Error(dotdotdotLoc, (Syntax_error msg)))
-      | None -> e
-      ) es in
+Solution: directly use `concat`." in
```

Comment:
Could you also change this one too:

```
Solution: directly use `concat` or other List helpers.
```

(I wrote it that way for List and Array because `concat` is shared by stdlib and Belt, but have different meanings. In reality you'd use `List.append` or `Belt.List.concat`, or `List.concat` or `Belt.List.concatMany`.)